opentelemetry-api/src/opentelemetry/resources/__init__.py

```diff
@@ -11,3 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+from abc import ABC, abstractmethod
+from typing import Dict, Union
+
+
+class Resource(ABC):
+    """This the interface that resources must implement"""
+    def __init__(self, labels: Dict[str, str]):
+        """Construct a resource.
+
+        Direct calling of the constructor is discouraged, as it cannot
+        take advantage of caching and restricts to the type of object
+        that can be returned.
+        """
+        self._labels = labels
+
+    @staticmethod
+    def create(labels: Dict[str, str]) -> "Resource":
+        """create a new resource.
+
+        Args:
+            labels: the labels that define the resource
+
+        Returns:
+            The resource with the labels in question
+        """
+    @property
+    @abstractmethod
+    def labels(self) -> Dict[str, str]:
+        """Return the label dictionary associated with this resource.
+
+        Returns:
+            A dictionary with the labels of the resource
+        """
+    def merge(self, other: Union["Resource", None]) -> "Resource":
+        """Return a resource with the union of labels for both resources.
+
+        Labels that exist in the main Resource take
```

Comment:
I know it doesn't make sense to say to follow PEP8 everywhere _except_ for the bit about wrapping docstring text at 72 chars, but having two different text widths in the same file seems like an unnecessary hassle for no obvious benefit.

Reply:
Sorry, it wasn't an intentional choice. Generally I use black which makes that consistent. I tried yapf which does not handle this reformatting.

I'll do some work to make sure I'm more consistent in the future.